### PR TITLE
[Messenger] ReceiverInterface updated

### PR DESCRIPTION
**Pull request by Guikingone**
Hy everyone 👋 , 

Due to 4.3 changements on `ReceiverInterface`, it seems that most of the existing code that we use is broken. 

It could be good to update the documentation according to the changements. 

This PR is linked to https://github.com/symfony/symfony-docs/issues/11789. 

Thanks for the feedback and have a great day 🙂

### CODE REVIEW

File: components/messenger.rst [hunk 1]
```diff
@@ -282,23 +282,37 @@ do is to write your own CSV receiver::
...
+        public function reject(Envelope $envelope): void
         {
-            // noop
+            // Reject the message if needed
         }
     }
 
+.. versionadded:: 4.3
+
+    The `ReceiverInterface` has been upgraded to act as most libraries
```
Comment (OskarStark) [inline components/messenger.rst]: Please 2 backticks instead of one.

Comment (Guikingone) [reply components/messenger.rst]: Fixed 🙂

**Review by OskarStark — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thanks Guillaume! We kindly thank you any help you can give us in Messenger-related issues and PRs.

File: components/messenger.rst [hunk 2]
```diff
@@ -282,23 +282,37 @@ do is to write your own CSV receiver::
...
         {
             $ordersFromCsv = $this->serializer->deserialize(file_get_contents($this->filePath), 'csv');
 
             foreach ($ordersFromCsv as $orderFromCsv) {
                 $order = new NewOrder($orderFromCsv['id'], $orderFromCsv['account_id'], $orderFromCsv['amount']);
+                
+                $envelope = new Envelope($order);
 
-                $handler(new Envelope($order));
+                $handler($envelope);
```
Comment (xabbuh) [inline components/messenger.rst]: Is this right? I don't see how `$handler` can be defined here.

Comment (Guikingone) [reply components/messenger.rst]: Oh, nice one 🤔

**Pull request merged**